runtime/src/iree/hal/drivers/cuda/cuda_driver.c

```diff
@@ -69,11 +97,23 @@ static iree_status_t iree_hal_cuda_driver_create_internal(
 
   iree_status_t status =
       iree_hal_cuda_dynamic_symbols_initialize(host_allocator, &driver->syms);
-  if (iree_status_is_ok(status)) {
-    *out_driver = (iree_hal_driver_t*)driver;
-  } else {
+  if (!iree_status_is_ok(status)) {
     iree_hal_driver_release((iree_hal_driver_t*)driver);
+    return status;
+  }
+
+#if IREE_HAL_DRIVER_CUDA_NCCL
+  // Initialize NCCL if NPROCS is set.
+  if (driver->default_params.nccl_default_count > 0) {
+    // get a unique ID from the environmental variable
+    status = iree_hal_nccl_get_unique_id_from_env(driver);
+    if (!iree_status_is_ok(status)) {
+      iree_hal_driver_release((iree_hal_driver_t*)driver);
+      return status;
+    }
   }
+#endif
```

Comment:
style nit (here and elsewhere): include the condition on the #endif `#endif  // IREE_HAL_DRIVER_CUDA_NCCL`